tools/list-not-released-prereleases.sh

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+ORG="newrelic";
+# https://docs.github.com/en/rest/teams/teams?apiVersion=2022-11-28#get-a-team-by-name
+# gh api /orgs/newrelic/teams/coreint | jq .id
+TEAM_ID="3626876";
+
+# Listing all repositories that start with `nri-` that belongs to coreint
+gh api --paginate "/teams/${TEAM_ID}/repos" --jq '.[] | select(.name | startswith("nri-")) | .name' | \
+while read REPO_NAME; do
+    # Get the date of the latest release.
+    LATEST_RELEASE=$(
+        gh release list --json publishedAt,isLatest -R "${ORG}/${REPO_NAME}" --jq '
+            .[] | select(.isLatest) | .publishedAt | fromdateiso8601
+        '
+    )
+
+    # List all releases, filter by prereleases that was made later than the latest release, sort by date,
+    # reverse because the last one is the most recent and only grab the last prerelease
+    NON_PUBLISHED_PRERELEASES=$(
+        gh release list --json name,publishedAt,isPrerelease,isLatest,tagName -R "${ORG}/${REPO_NAME}" --jq '
+            [
+                .[] | select( .isPrerelease and ((.publishedAt | fromdateiso8601) > '${LATEST_RELEASE}'))
+            ]
+            | sort_by(.publishedAt |= fromdateiso8601) | reverse | first
+        '
+    )
+    if ! [ -z "${NON_PUBLISHED_PRERELEASES}" ]; then
+        # Next querie uses the list release API endpoint:
+        # https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28#list-releases
+
+        # gh list release is limited only to fields createdAt, isDraft, isLatest, isPrerelease, name, publishedAt, and tagName
+        # We need the URL to show it to the user.
+
+        NAME=$(jq -r .name <<<"${NON_PUBLISHED_PRERELEASES}") # Name of the release
+        TAG=$(jq -r .tagName <<<"${NON_PUBLISHED_PRERELEASES}") # Tag of the release
+        URL=$(gh api --paginate "/repos/${ORG}/${REPO_NAME}/releases/tags/${TAG}" --jq '.html_url') # URL of the release
+
+        echo " > Prerelease found for ${REPO_NAME}: ${URL}"
```

Comment:
I wonder if we could algo get here in any run for this release failed (to avoid promoting by mistake a pre-release that failed). I guess we would need to define the _status check_ from `list-failed-releases.sh` in some sort of function 🤔

Reply:
I solved on [88a05cc](https://github.com/newrelic/coreint-automation/pull/59/commits/88a05cc4de145dadbac8331fa02e50429edd8fa9). I introduced also a fez changes so I edited the PR description.

Let me know what you think.